#163518658 Search & Filter Articles

abayo-luc opened the pull request:
#### What does this PR do?
Search Article by keyword
#### Description of Task to be completed?
- Convert HTML & CSS mock into React component
- Fetchs article and store them into Redux state
- Displies fetchec article 
- Test: component, reducers, and action creators
- Refactors app routing
#### How should this be manually tested?
- clone the repo and check out the following branch `ft-search-and-filtering-163518658 `
- run `yarn start` and navigate to / in your browser, type in a search query to be redirected to search results page
### Related Pivotal tracker ID
#163518658

YvesIraguha committed:
chore(create):create eslint file

-enable hound ci
[starts #165054398]

chore(fix): ignore unnecessary files

chore(fix): remove unneccessary files

[#164364254]Add search results page

[#164364254]Add nav-bar to search page

[#164364254]Remove broken tabs

feat(user setting):Add user settings pages- Update password- Mute or unmute email notification- Delete my account

[#164203446] add landing page -  desktop view

[#164203446] add landing page -  responsive complete

[feature #164395296] refactor sass

[ch #[163518639]] create route for the application

[ch #[163518639]] fix issue where link return 404

[ft #[163518639]] remove navigation bar

[ch #[163518639]] refractor the code by putting routes in App.js

[ch #[163518639]]  update .gitignore file

espoirMur committed:
[ch #[163518639]] fix rebase conflict from develop

espoirMur committed:
[ft #[163518639]] fix test faling issues

niomwungeri-fabrice committed:
add profile view, edit and settings

abayo-luc committed:
[Feature #163518658] article search & filter

- Convert HTML & CSS mock into React component
- Fetchs article and store them into Redux state
- Displies fetchec article
- Test: component, reducers, and action creators
- Refactors app routing

[Finishes #163518658]

abayo-luc committed:
[Feature #163518658] article search & filter

- Convert HTML & CSS mock into React component
- Fetchs article and store them into Redux state
- Displies fetchec article
- Test: component, reducers, and action creators
- Refactors app routing

[Finishes #163518658]

abayo-luc committed:
[Feature #163518658] article search & filter

- Convert HTML & CSS mock into React component
- Fetchs article and store them into Redux state
- Displies fetchec article
- Test: component, reducers, and action creators
- Refactors app routing

[Finishes #163518658]

abayo-luc committed:
[Feature #163518658] article search & filter

- Convert HTML & CSS mock into React component
- Fetchs article and store them into Redux state
- Displies fetchec article
- Test: component, reducers, and action creators
- Refactors app routing

[Finishes #163518658]

abayo-luc committed:
[Feature #163518658] article search & filter

- Convert HTML & CSS mock into React component
- Fetchs article and store them into Redux state
- Displies fetchec article
- Test: component, reducers, and action creators
- Refactors app routing

[Finishes #163518658]

abayo-luc committed:
[Feature #163518658] article search & filter

- Convert HTML & CSS mock into React component
- Fetchs article and store them into Redux state
- Displies fetchec article
- Test: component, reducers, and action creators
- Refactors app routing

[Finishes #163518658]

abayo-luc committed:
[Feature #163518658] article search & filter

- Convert HTML & CSS mock into React component
- Fetchs article and store them into Redux state
- Displies fetchec article
- Test: component, reducers, and action creators
- Refactors app routing

[Finishes #163518658]

abayo-luc committed:
[Feature #163518658] article search & filter

- Convert HTML & CSS mock into React component
- Fetchs article and store them into Redux state
- Displies fetchec article
- Test: component, reducers, and action creators
- Refactors app routing

[Finishes #163518658]

abayo-luc committed:
[feature #163518658] rebases develop

abayo-luc committed:
[feature #163518658] rebases develop

abayo-luc committed:
[feature #163518658] rebases develop

abayo-luc committed:
[feature #163518658] rebases develop

abayo-luc committed:
[Feature #163518658] clear search results on component will unmount

abayo-luc committed:
[Feature #163518658] clear search results on component will unmount

abayo-luc committed:
[Feature #163518658] clear search results on component will unmount

abayo-luc committed:
[Feature #163518658] clear search results on component will unmount

abayo-luc committed:
[Feature #163518658] clear search results on component will unmount

abayo-luc committed:
[Feature #163518658] clear search results on component will unmount

abayo-luc committed:
[Feature #163518658] clear search results on component will unmount

abayo-luc committed:
[Feature #163518658] clear search results on component will unmount

abayo-luc committed:
[Feature #163518658] add test for class methods

abayo-luc committed:
[Feature #163518658] add test for class methods

abayo-luc committed:
[Feature #163518658] add test for class methods

abayo-luc committed:
[Feature #163518658] add test for class methods

abayo-luc committed:
[Feature #163518658] add test for class methods

abayo-luc committed:
[Feature #163518658] add test for class methods

abayo-luc committed:
[Feature #163518658] adds onKeyDown event on search text input

abayo-luc committed:
[Feature #163518658] adds onKeyDown event on search text input

abayo-luc committed:
[Feature #163518658] adds onKeyDown event on search text input

abayo-luc committed:
[Feature #163518658] adds onKeyDown event on search text input

abayo-luc committed:
[Feature #163518658] adds onKeyDown event on search text input

abayo-luc committed:
[Feature #163518658] adds onKeyDown event on search text input

abayo-luc committed:
[Feature #163518658] adds onKeyDown event on search text input

abayo-luc committed:
[Feature #163518658] remove console logs

abayo-luc committed:
[Feature #163518658] add loading animation and error message

- Designs loading animation with SCSSS
- Creates loading components
- Display loading animation or errors message on SearchResults page

abayo-luc committed:
[Feature #163518658] add loading animation and error message

- Designs loading animation with SCSSS
- Creates loading components
- Display loading animation or errors message on SearchResults page

abayo-luc committed:
[Feature #163518658] add loading animation and error message

- Designs loading animation with SCSSS
- Creates loading components
- Display loading animation or errors message on SearchResults page

abayo-luc committed:
[Feature #163518658] add loading animation and error message

- Designs loading animation with SCSSS
- Creates loading components
- Display loading animation or errors message on SearchResults page

abayo-luc committed:
[Feature #163518658] add loading animation and error message

- Designs loading animation with SCSSS
- Creates loading components
- Display loading animation or errors message on SearchResults page

abayo-luc committed:
[Feature #163518658] add loading animation and error message

- Designs loading animation with SCSSS
- Creates loading components
- Display loading animation or errors message on SearchResults page

abayo-luc committed:
[Feature #163518658] add loading animation and error message

- Designs loading animation with SCSSS
- Creates loading components
- Display loading animation or errors message on SearchResults page

abayo-luc committed:
[Feature #163518658] add loading animation and error message

- Designs loading animation with SCSSS
- Creates loading components
- Display loading animation or errors message on SearchResults page

abayo-luc committed:
[Feature #163518658] updates the search retuls page snapshot

abayo-luc committed:
[Feature #163518658] updates the search retuls page snapshot

abayo-luc committed:
[Feature #163518658] updates the search retuls page snapshot

abayo-luc committed:
[Feature #163518658] updates the search retuls page snapshot

abayo-luc committed:
[Feature #163518658] updates the search retuls page snapshot

abayo-luc committed:
[Feature #163518658] updates the search retuls page snapshot

abayo-luc committed:
[Feature #163518658] updates the search retuls page snapshot

abayo-luc committed:
[Feature #163518658] updates the search retuls page snapshot

abayo-luc committed:
[Feature #163518658] code refactor

abayo-luc committed:
[Feature #163518658] code refactor

abayo-luc committed:
[Feature #163518658] code refactor

abayo-luc committed:
[Feature #163518658] code refactor

abayo-luc committed:
[Chore #163518658] remove snapshots from git

abayo-luc committed:
[Chore #163518658] remove snapshots from git

abayo-luc committed:
[Chore #163518658] remove snapshots from git

abayo-luc committed:
[Chore #163518658] remove snapshots from git

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

abayo-luc committed:
[Feature #163518658] add popover card for suggested articles

YvesIraguha committed:
chore(fix):  reporting form
- Create reporting form
- Fix reactions buttons on scroll

abayo-luc committed:
[Feature #163518658] fixing autosuggested articles on search input change

abayo-luc committed:
[Feature #163518658] fixing autosuggested articles on search input change

abayo-luc committed:
[Feature #163518658] fixing autosuggested articles on search input change

abayo-luc committed:
[Feature #163518658] fixing autosuggested articles on search input change

abayo-luc committed:
[Feature #163518658] fixing autosuggested articles on search input change

abayo-luc committed:
[Feature #163518658] fixing autosuggested articles on search input change

abayo-luc committed:
[Feature #163518658] fixing autosuggested articles on search input change

abayo-luc committed:
[Feature #163518658] fixing autosuggested articles on search input change

abayo-luc committed:
[Feature #163518658] tests infinit scroll feature

abayo-luc committed:
[Feature #163518658] tests infinit scroll feature

abayo-luc committed:
[Feature #163518658] tests infinit scroll feature

abayo-luc committed:
[Feature #163518658] tests infinit scroll feature

abayo-luc committed:
[Feature #163518658] tests infinit scroll feature

abayo-luc committed:
[Feature #163518658] tests infinit scroll feature

abayo-luc committed:
[Feature #163518658] tests infinit scroll feature

abayo-luc committed:
[Feature #163518658] tests infinit scroll feature

abayo-luc committed:
Fixes houndCI warnings

abayo-luc committed:
Fixes houndCI warnings

abayo-luc committed:
Fixes houndCI warnings

abayo-luc committed:
Fixes houndCI warnings

abayo-luc committed:
Fixes houndCI warnings

abayo-luc committed:
Fixes houndCI warnings

abayo-luc committed:
Fixes houndCI warnings

abayo-luc committed:
Update snapshot on travis

abayo-luc committed:
Update snapshot on travis

abayo-luc committed:
Update snapshot on travis

abayo-luc committed:
Update snapshot on travis

abayo-luc committed:
Update snapshot on travis

abayo-luc committed:
Update snapshot on travis

abayo-luc committed:
Update snapshot on travis

abayo-luc committed:
Update snapshot on travis

abayo-luc committed:
Update snapshot on travis

abayo-luc committed:
Update snapshot on travis

abayo-luc committed:
Update snapshot on travis

abayo-luc committed:
[Feature #163518658] fixes eslint errors

abayo-luc committed:
[Feature #163518658] fixes eslint errors

abayo-luc committed:
[Feature #163518658] fixes eslint errors

abayo-luc committed:
[Feature #163518658] fixes eslint errors

abayo-luc committed:
Rebase Develop

abayo-luc committed:
Rebase Develop

abayo-luc committed:
Rebase Develop

abayo-luc committed:
Rebase Develop

abayo-luc committed:
Fixes travis

abayo-luc committed:
Fixes travis

abayo-luc committed:
Fixes travis

abayo-luc committed:
Fixes travis

YvesIraguha commented on src/components/PopOvers/SearchPopOver.js:
```diff
@@ -0,0 +1,52 @@
+
+const SearchPopOver = ({ searchQuery, articles }) => (
+  <div id="search-popover">
+    <div className="popover-inner" style={{ height: "260px" }}>
```
@abayo-luc why inline styles here?

YvesIraguha commented on src/components/PopOvers/SearchPopOver.js:
```diff
@@ -0,0 +1,52 @@
+          </Link>
+        </div>
+      </header>
+      <ul className="popover-content" style={{ maxHeight: "250px" }}>
```
inline styles?

YvesIraguha commented on src/components/common/AppBars/navBar.js:
```diff
@@ -104,9 +131,41 @@ class Navbar extends Component {
+  return {
+    ...search
+  };
+};
```
I think you can use object shorthand destructuring, even more, here. => `({...state.search})`

abayo-luc replied:
Nice one here !!

YvesIraguha commented on src/components/common/Cards/AnimatedCard.js:
```diff
@@ -0,0 +1,105 @@
+          style={{
+            backgroundImage: `url(${require("../../../assets/img/no-thumbnail.png")})`
+          }}
+        />
```
We can keep using es6 for importing images.

YvesIraguha commented on src/components/common/Cards/AnimatedCard.js:
```diff
@@ -0,0 +1,105 @@
+            src={author.image || require("../../../assets/img/user.png")}
+            alt="user"
+            className="user-profile-img"
+          />
```
We can keep using es6 for importing images.

YvesIraguha commented on src/components/common/Cards/AnimatedCard.js:
```diff
@@ -0,0 +1,105 @@
+                src={require("../../../assets/img/like-icon.svg")}
+                alt="like"
+                className="icon-like"
+              />
```
We can keep using es6 for importing images.

YvesIraguha commented on src/components/common/Cards/Latest.js:
```diff
@@ -33,27 +33,27 @@ export default class Latest extends Component {
             <div className="left-icons">
               <img
-                src={require("../../../assets/img/thumb-up-outline.svg")}
+                src={require("../../../assets/img/like-icon.svg")}
```
check comments above on using ES6.

abayo-luc committed:
[Feature #163518658] improves the user exp

- Cneters loading animation
- Add separator between suggested article
- Add author's username and read time on suggested articles
- Simulates not found articles on user's enter special char (to be remove)
- Improve no results found page

[Delivers #163518658]

abayo-luc committed:
[Feature #163518658] improves the user exp

- Cneters loading animation
- Add separator between suggested article
- Add author's username and read time on suggested articles
- Simulates not found articles on user's enter special char (to be remove)
- Improve no results found page

[Delivers #163518658]

abayo-luc committed:
[Feature #163518658] improves the user exp

- Cneters loading animation
- Add separator between suggested article
- Add author's username and read time on suggested articles
- Simulates not found articles on user's enter special char (to be remove)
- Improve no results found page

[Delivers #163518658]

abayo-luc committed:
[Feature #163518658] improves the user exp

- Cneters loading animation
- Add separator between suggested article
- Add author's username and read time on suggested articles
- Simulates not found articles on user's enter special char (to be remove)
- Improve no results found page

[Delivers #163518658]

abayo-luc committed:
Fixes infinit scroll

abayo-luc committed:
Fixes infinit scroll

abayo-luc committed:
Fixes infinit scroll

abayo-luc committed:
Fixes infinit scroll

abayo-luc committed:
remove style duplicate

YvesIraguha commented on src/__tests__/__actions__/searchActions.test.js:
```diff
@@ -0,0 +1,165 @@
+    });
+  });
+
+  test("should dispatch ARTICLE_SEARCH_FAILED ", () => {
```
Hello @luc in `authAction.test.js` you used `it()` for wrapping a test case, here you are using `test()`. I know that the result is similar but shouldn't we stick to consistency?

abayo-luc replied:
I think you are making sense. But test() have been used in so many places. I am not actually the only one using it. I think we should raise this concern during the StandUp for today.

YvesIraguha commented on src/components/common/AppBars/navBar.js:
```diff
@@ -1,75 +1,104 @@
+    getSuggestions(value);
+  };
+
   toggleOptions() {
```
For the sake of consistency, this should be an allow function.

YvesIraguha commented on src/components/common/Cards/AuthorCard.js:
```diff
@@ -0,0 +1,40 @@
+      <div className="inline-author">
+        <div className="user-profile-img">
+          <img
+            src={image || require("../../../assets/img/user.png")}
```
@abayo-luc use ES6 for importing images.

YvesIraguha commented on src/redux/actions/searchActions.js:
```diff
@@ -0,0 +1,86 @@
+    const authorsObject = mapValues(
+      arrayToObject(articles, "userId"),
+      article => ({ ...article.author, id: article.userId })
+    );
```
Kudos! @abayo-luc

YvesIraguha commented on src/utils/helperFunctions.js:
```diff
@@ -20,3 +23,66 @@ export const capitalize = string =>
+  return date.toDateString();
+};
+
+export const arrayToObject = (items, key) => keyBy(items, item => item[key]);
```
JS-doc for this function and functions below is required.

YvesIraguha reviewed (changes requested):
@abayo-luc I made some comments in different files. Overall, you did a great job. However, in future check closely on the naming of some variables, and files. Names are correct, but they have typos. For instance, instead of `SearchResult.test.js` you wrote `SearchRetuls.test.js`, instead of `wrapper` you wrote `warper`. You can look closely to avoid other instances in the future.

abayo-luc committed:
[Feature #163518658] reaplaces require with import

abayo-luc committed:
[Feature #163518658] reaplaces require with import

abayo-luc committed:
[Feature #163518658] reaplaces require with import

abayo-luc committed:
[Feature #163518658] reaplaces require with import

abayo-luc committed:
[Feature #163518658] rebases develop

abayo-luc committed:
[Feature #163518658] rebases develop

abayo-luc committed:
[Feature #163518658] rebases develop

abayo-luc committed:
[Feature #163518658] rebases develop

abayo-luc committed:
[Feature #163518658] fixes pop-over height

abayo-luc committed:
[Feature #163518658] fixes pop-over height

abayo-luc committed:
[Feature #163518658] fixes pop-over height

abayo-luc committed:
[Feature #163518658] fixes pop-over height

abayo-luc committed:
fixes .gitignore

abayo-luc committed:
fixes .gitignore

abayo-luc committed:
fixes .gitignore

abayo-luc committed:
[Feature #163518658] code refactoring

abayo-luc committed:
[Feature #163518658] code refactoring

abayo-luc committed:
[Feature #163518658] code refactoring

abayo-luc committed:
[Feature #163518658] code refactoring

abayo-luc committed:
[Feature #163518658]Add time out on authsuggesting articles

abayo-luc committed:
[Feature #163518658]Add time out on authsuggesting articles

abayo-luc committed:
[Feature #163518658]Add time out on authsuggesting articles

abayo-luc committed:
[Feature #163518658] adds more test to improve test coverage

abayo-luc committed:
[Feature #163518658] adds more test to improve test coverage

abayo-luc committed:
[Feature #163518658] adds more test to improve test coverage

abayo-luc committed:
[Feature #163518658] adds more test to improve test coverage

abayo-luc committed:
[Feature #163518658] fixes redirect

abayo-luc committed:
[Feature #163518658] fixes redirect

abayo-luc committed:
[Feature #163518658] fixes redirect

oesukam committed:
Merge pull request #37 from andela/ft-view-edit-profile-#163518646

163518646  View and Edit Profile

oesukam commented on package.json:
```diff
@@ -41,8 +41,10 @@
     "dotenv": "^7.0.0",
     "jest": "^24.7.1",
     "jsonwebtoken": "^8.5.1",
+    "lodash": "^4.17.11",
```
Why do you need lodash?

abayo-luc replied:
I am dealing with object and array so many times. Without Lodash, I would end up writing so many custom functions and repeating myself.

oesukam commented on src/__tests__/__actions__/searchActions.test.js:
```diff
@@ -0,0 +1,195 @@
+      }
+    ];
+    moxios.stubRequest(
+      `${process.env.API_BASE_URL}/articles?keyword=${keyword}&page=${2}`,
```
@abayo-luc You should destruct API_BASE_URL

oesukam commented on src/__tests__/__actions__/searchActions.test.js:
```diff
@@ -0,0 +1,195 @@
+      },
+      {
+        type: SET_SUGGESTED_ARTICLES,
+        payload: { articles: { ...arrayToObject(articles, "id") } }
```
@abayo-luc  Can't be done with native functions?

abayo-luc replied:
I can write a custom function for that. But still, I need Lodash to prevent me from writing more than 10 custom functions.

oesukam commented on src/__tests__/__components__/AppBars/CategoryBar.test.js:
```diff
@@ -0,0 +1,30 @@
+describe("Category Bar Component", () => {
+  test("should match the snapshot", () => {
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
```
Empty line required

oesukam commented on src/__tests__/__components__/AppBars/NavBar.test.js:
```diff
@@ -0,0 +1,85 @@
+      jest.spyOn(instance, "handleEnterPress");
+      jest.spyOn(instance, "closeSearchPopOver");
+      jest.spyOn(instance, "handleOnChange");
+    });
```
Empty line required

oesukam commented on src/__tests__/__components__/AppBars/NavBar.test.js:
```diff
@@ -0,0 +1,85 @@
+      instance.handleEnterPress.mockClear();
+      instance.closeSearchPopOver.mockClear();
+      instance.handleOnChange.mockClear();
+    });
```
Empty line required

oesukam commented on src/__tests__/__reducers__/searchReducers.test.js:
```diff
@@ -0,0 +1,92 @@
+describe("Search Action Reducers", () => {
+  test("should should match the initial state", () => {
+    expect(searchReducers(undefined, {})).toEqual({ ...INITIAL_STATE });
+  });
```
Empty line required

oesukam commented on src/__tests__/__views__/Home.test.js:
```diff
@@ -2,10 +2,13 @@ import React from "react";
-    expect(warper.find("h2").text()).toEqual("Hello world, from Titan-Devs");
+    const warper = mount(<App {...props} />);
+    expect(warper.find("h2").text()).toEqual("Hello home");
     expect(true).toBe(true);
```
What are you trying to test?

oesukam commented on src/__tests__/__views__/SearchRetuls.test.js:
```diff
@@ -0,0 +1,270 @@
+describe("Search Results Component", () => {
+  afterEach(() => {
+    wrapper.setProps({
+      ...props
```
Why are you passing again props yet you did the same thing on line 34?

oesukam commented on src/components/Animations/LoadingDots.js:
```diff
@@ -0,0 +1,25 @@
+const BottomPopUp = () => (
+  <div className="lds-css ng-scope loading">
+    <div style={{ width: "100%", height: "100%" }} className="lds-ellipsis">
+      <div>
```
@abayo-luc Can't you generate all those divs as they are similar?

oesukam commented on src/components/PopOvers/SearchPopOver.js:
```diff
@@ -0,0 +1,55 @@
+SearchPopOver.propTypes = {
+  searchQuery: PropTypes.string,
+  articles: PropTypes.shape({}).isRequired
+};
```
Empty line required

oesukam commented on src/components/PopOvers/SearchPopOver.js:
```diff
@@ -0,0 +1,55 @@
+};
+SearchPopOver.defaultProps = {
+  searchQuery: ""
+};
```
Empty line required

oesukam commented on src/components/common/AppBars/navBar.js:
```diff
@@ -1,40 +1,98 @@
+                    placeholder="Search...."
+                    onChange={e => this.handleOnChange(e.target.value)}
+                    value={searchQuery}
+                    onKeyDown={e => this.handleEnterPress(e)}
```
Just pass the reference to the function instead of using the arrow function

oesukam commented on src/components/common/AppBars/navBar.js:
```diff
@@ -1,40 +1,98 @@
+                    type="search"
+                    name="search"
+                    placeholder="Search...."
+                    onChange={e => this.handleOnChange(e.target.value)}
```
Can't you just use the function's reference?

oesukam commented on src/components/common/AppBars/navBar.js:
```diff
@@ -77,9 +135,38 @@ class Navbar extends Component {
             </div>
           </div>
         </section>
+        {history.location.pathname !== "/search" &&
```
pathname should be destructed

oesukam commented on src/components/common/AppBars/navBar.js:
```diff
@@ -77,9 +135,38 @@ class Navbar extends Component {
         </section>
+        {history.location.pathname !== "/search" &&
+          popOverOpen &&
+          !isEmpty(suggestedArticles) && (
```
Why using `isEmpty`?

oesukam commented on src/components/common/AppBars/navBar.js:
```diff
@@ -77,9 +135,38 @@ class Navbar extends Component {
+    push: PropTypes.func
+  }).isRequired,
+  suggestedArticles: PropTypes.shape({})
+};
```
Empty line required

oesukam commented on src/components/common/Cards/AnimatedCard.js:
```diff
@@ -0,0 +1,95 @@
+        <h2>{title}</h2>
+      </div>
+      <p className="card-text">
+        {description}......
```
Why those dots?

oesukam commented on src/components/common/Cards/AnimatedCard.js:
```diff
@@ -0,0 +1,95 @@
+  readTime: PropTypes.number.isRequired,
+  createdAt: PropTypes.string.isRequired,
+  slug: PropTypes.string
+};
```
Empty line required

oesukam commented on src/components/common/Cards/AnimatedCard.js:
```diff
@@ -0,0 +1,95 @@
+};
+AnimatedCard.defaultProps = {
+  slug: ""
+};
```
Empty line required

oesukam commented on src/components/common/Cards/AuthorCard.js:
```diff
@@ -0,0 +1,38 @@
+  following: PropTypes.bool,
+  image: PropTypes.string,
+  username: PropTypes.string.isRequired
+};
```
Empty line required

oesukam commented on src/components/common/Cards/AuthorCard.js:
```diff
@@ -0,0 +1,38 @@
+AuthorCard.defaultProps = {
+  following: false,
+  image: null
+};
```
Empty line required

oesukam commented on src/components/common/Inputs/TextInput.js:
```diff
@@ -27,10 +27,19 @@ TextInput.propTypes = {
-  value: PropTypes.string.isRequired
+  value: PropTypes.string,
+  onKeyDown: PropTypes.func
 };
```
Empty line required

oesukam commented on src/redux/actions/authActions.js:
```diff
@@ -21,6 +21,10 @@ export const handleTextInput = (name, value) => ({
 const updateIsSubmitting = () => ({
   type: SUBMITTING_LOGIN_CREDENTIALS
 });
```
Empty line required

oesukam commented on src/redux/actions/searchActions.js:
```diff
@@ -0,0 +1,85 @@
+
+export const clearSearchResults = () => ({
+  type: CLEAR_SEARCH_RESULTS
+});
```
Empty line required

Musigwa-zz commented on src/__tests__/__actions__/searchActions.test.js:
```diff
@@ -0,0 +1,195 @@
+      },
+      {
+        type: SET_SUGGESTED_ARTICLES,
+        payload: { articles: { ...arrayToObject(articles, "id") } }
```
@abayo-luc, if the `arrayToObject()` returns an object, there is no need to spread it in an object. I think that you should rather assign its return value directly to the `articles` unless you have some additional properties that you need to give your `articles` object.

Musigwa-zz commented on src/__tests__/__actions__/searchActions.test.js:
```diff
@@ -0,0 +1,195 @@
+      {
+        status: 500,
+        response: {
+          ...payload
```
I think it would be better if you assigned the payload to the response object unless you have some other properties which you need to give it.

Musigwa-zz commented on src/__tests__/__components__/AppBars/NavBar.test.js:
```diff
@@ -0,0 +1,85 @@
+jest.useFakeTimers();
+describe("NavBar component", () => {
+  describe("rendered component", () => {
+    test("should match the snapshot", () => {
```
Please consider leaving an empty space between your tests for better readability.

Musigwa-zz commented on src/components/common/AppBars/CategoryBar.js:
```diff
@@ -24,13 +24,18 @@ class Categories extends Component {
         </div>
-        <div onClick={() => onMoreClick()} className="category-right">
+        <div
+          onClick={() => onMoreClick()}
```
What do you think about passing your click handler as a reference instead?

abayo-luc replied:
Personally, I don't prefer references if there is a way to declary function and use it directly. 

<img width="944" alt="Screen Shot 2019-05-07 at 10 58 21" src="https://user-images.githubusercontent.com/20681465/57287162-30b4eb80-70b7-11e9-87fb-3590d5dd6ed9.png">

Musigwa-zz commented on src/views/SearchResults.js:
```diff
@@ -0,0 +1,241 @@
+              className="search-input"
+              onChange={e => this.handleOnChange(e.target.value)}
+              value={searchQuery}
+              onKeyDown={e => this.handleEnterPress(e)}
```
Should be the function reference instead.

abayo-luc replied:
<img width="944" alt="Screen Shot 2019-05-07 at 10 58 21" src="https://user-images.githubusercontent.com/20681465/57287107-13801d00-70b7-11e9-9f20-f45645066c6a.png">

Musigwa-zz reviewed (commented):
Good work Mr. @abayo-luc. Please consider some comments I left here.

abayo-luc committed:
[ Feature #163518658] replace lodash with custom functions

abayo-luc committed:
[ Feature #163518658] replace lodash with custom functions

abayo-luc committed:
[ Feature #163518658] replace lodash with custom functions

abayo-luc committed:
[ Feature #163518658] replace lodash with custom functions

abayo-luc committed:
[Feature #163518658] add empty line between test cases

abayo-luc committed:
[Feature #163518658] add empty line between test cases

abayo-luc committed:
[Feature #163518658] add empty line between test cases

abayo-luc committed:
[Feature #163518658] add empty line between test cases

abayo-luc committed:
[Feature #163518658] add empty line between test cases

YvesIraguha committed:
chore(fix): add changes from edit profile

abayo-luc committed:
[Feature #163518658] rebase develop

abayo-luc committed:
[Feature #163518658] rebase develop

abayo-luc committed:
[Feature #163518658] fix loading animation

- Styles loading animation
- Add more test to for on scroll event
- Rebase and fixes merge conflict

[Delivers #163518658]

abayo-luc committed:
[Feature #163518658] fix loading animation

- Styles loading animation
- Add more test to for on scroll event
- Rebase and fixes merge conflict

[Delivers #163518658]

abayo-luc committed:
[Feature #163518658] fix loading animation

- Styles loading animation
- Add more test to for on scroll event
- Rebase and fixes merge conflict

[Delivers #163518658]

abayo-luc committed:
[Feature #163518658] fix loading animation

- Styles loading animation
- Add more test to for on scroll event
- Rebase and fixes merge conflict

[Delivers #163518658]

abayo-luc committed:
[Feature #163518658] fix loading animation

- Styles loading animation
- Add more test to for on scroll event
- Rebase and fixes merge conflict

[Delivers #163518658]

abayo-luc committed:
[Feature #163518658] replaces warper with wrapper

abayo-luc committed:
[Feature #163518658] replaces warper with wrapper

abayo-luc committed:
[Feature #163518658] replaces warper with wrapper

abayo-luc committed:
[Feature #163518658] replaces warper with wrapper

abayo-luc committed:
[Feature #163518658] replaces yarn with npm on travis

abayo-luc committed:
[Feature #163518658] replaces yarn with npm on travis

abayo-luc committed:
Update .travis.yml

YvesIraguha committed:
fix heights lenght

YvesIraguha committed:
chore(fix): use destructuing tests

YvesIraguha committed:
chore(fix): use destructuing for env varaibles

oesukam committed:
Merge pull request #52 from andela/ch-fix-reactions-buttons-on-scoll-165716231

#165716231 Fix Reactions Buttons on Scroll

abayo-luc committed:
Resolve merge conflict

abayo-luc committed:
Fixes read article: login reducer to auth

abayo-luc committed:
[Featue #163518658] fixes read article tests

abayo-luc committed:
Update travis cofig

abayo-luc committed:
Update travis config
